### PR TITLE
[AVR-881] added distance to reference trajectory metrics

### PR DESCRIPTION
**Pull request by yawei-ye**
# Changes
- [x] added distance to reference trajectory metric
- [x] added unit test for new metric

### CODE REVIEW

File: l5kit/l5kit/evaluation/metrics.py [hunk 1]
```diff
@@ -303,3 +304,22 @@ def final_displacement_error_mean(
     """
 
     return _final_displacement_error(ground_truth, pred, confidences, avails, "mean")
+
+
+def distance_to_reference_trajectory(pred_centroid: torch.Tensor, ref_traj: torch.Tensor) -> torch.Tensor:
```
Comment (lucabergamini) [inline l5kit/l5kit/evaluation/metrics.py:309]: #226 works with numpy array, this one with tensors. Nothing against it if we're refactoring them later anyway

File: l5kit/l5kit/evaluation/metrics.py [hunk 2]
```diff
@@ -303,3 +304,22 @@ def final_displacement_error_mean(
...
+    :type ref_traj: torch.Tensor, float
+    :return: closest distance between the predicted centroid and the reference trajectory, size: [batch_size,]
+    :rtype: torch.Tensor, float
+    """
+    # [batch_size, 2]
+    assert pred_centroid.dim() == 2
+    # [batch_size, num_timestamps, 2]
+    assert ref_traj.dim() == 3
+
+    error, _ = torch.min(torch.norm(pred_centroid.unsqueeze(1) - ref_traj, p=2, dim=-1), dim=1)
```
Comment (lucabergamini) [inline l5kit/l5kit/evaluation/metrics.py:324]: you can use `values` if you only want that (and not the indices)

File: l5kit/l5kit/tests/evaluation/metrics_test.py [hunk 3]
```diff
@@ -141,3 +142,19 @@ def test_ade_fde_known_results() -> None:
...
     assert np.allclose(final_displacement_error_oracle(gt, pred, confs, avail), 0.5)
+
+
+@pytest.mark.parametrize("device", ["cpu", "cuda"])
+def test_distance_to_reference_trajectory(device: str) -> None:
+    if device == "cuda" and not torch.cuda.is_available():
+        pytest.skip("requires CUDA")
+
+    # [batch_size, 2]
+    pred_centroid = torch.tensor([[1, 1], [1.5, 2]], dtype=torch.float32, device=device)
```
Comment (lucabergamini) [inline l5kit/l5kit/tests/evaluation/metrics_test.py:153]: I would add the trivial case where the point is on the trajectory

**Review by lucabergamini — DISMISSED** [review]
minor but LGTM

File: l5kit/l5kit/evaluation/metrics.py (hunk 1 above)
Comment (yawei-ye) [reply l5kit/l5kit/evaluation/metrics.py:309]: I would like to compute the metrics using GPU, so I would use tensor whenever possible

**Comment by yawei-ye** [comment]
please take another look @lucabergamini @perone

**Comment by perone** [comment]
Sorry @yawei-ye I just received a notification now, will take a look !

File: l5kit/l5kit/evaluation/metrics.py [hunk 4]
```diff
@@ -303,3 +304,23 @@ def final_displacement_error_mean(
...
+    :return: closest distance between the predicted centroid and the reference trajectory, size: [batch_size,]
+    :rtype: torch.Tensor, float
+    """
+    # [batch_size, 2]
+    assert pred_centroid.dim() == 2
+    # [batch_size, num_timestamps, 2]
+    assert ref_traj.dim() == 3
+
+    # [batch_size,]
+    error = torch.min(torch.norm(pred_centroid.unsqueeze(1) - ref_traj, p=2, dim=-1), dim=1).values
```
Comment (perone) [inline l5kit/l5kit/evaluation/metrics.py:325]:
```suggestion
    euclidean_distance = torch.norm(pred_centroid.unsqueeze(1) - ref_traj, p=2, dim=-1)
    return torch.amin(euclidean_distance, dim=1)
```


PS: `torch.norm` is being deprecated in favor or `torch.linalg.norm`.

**Review by perone — DISMISSED** [review]
Just minor, but consider the `amin` instead of `min` if you don't need the indices.

**Comment by yawei-ye** [comment]
> Just minor, but consider the `amin` instead of `min` if you don't need the indices.

good suggestion, I have made the change

**Review by lucabergamini — APPROVED** [review]
